src/gsi/setupdbz.f90

```diff
@@ -1928,29 +1930,37 @@ subroutine contents_netcdf_diag_(odiag)
            call nc_diag_metadata("Observation_Class",       obsclass            )
            call nc_diag_metadata("Observation_Type",        ictype(ikx)         )
            call nc_diag_metadata("Observation_Subtype",     icsubtype(ikx)      )
-           call nc_diag_metadata("Latitude",                sngl(data(ilate,i)) )
-           call nc_diag_metadata("Longitude",               sngl(data(ilone,i)) )
-           call nc_diag_metadata("Station_Elevation",       sngl(data(ielev,i)) )
-           call nc_diag_metadata("Pressure",                sngl(presw)         )
-           call nc_diag_metadata("Height",                  sngl(data(ihgt,i))  )
-           call nc_diag_metadata("Time",                    sngl(dtime-time_offset))
-           call nc_diag_metadata("Prep_QC_Mark",            sngl(zero)          )
-           call nc_diag_metadata("Prep_Use_Flag",           sngl(data(iuse,i))  )
+           call screen_to_single_nc_diag_metadata("Latitude",(data(ilate,i)) )
+           call screen_to_single_nc_diag_metadata("Longitude",(data(ilone,i)) )
+           call screen_to_single_nc_diag_metadata("Station_Elevation",(data(ielev,i)) )
+           call screen_to_single_nc_diag_metadata("Pressure",(presw)         )
+           call screen_to_single_nc_diag_metadata("Height",(data(ihgt,i))  )
+           if(isnan(dtime) .or. isnan(time_offset)) then
```

Comment:
Still confused why we need to set `r_missing` if NaN is present and then call `screen_to_single_nc_diag_metadata`.   Doesn't `screen_to_single_nc_diag_metadata` do the same thing?

Reply:
No, I don't thing we should.  I've teetered on how best to resolve this and have decided to write another subroutine in ncdiag to screen both variables and perform the calculation.  I will include this in the PR addressing NOAA-EMC/GSI-ncdiag#1.

Reply:
Fixed.